ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/DefaultBlockchain.java

```diff
@@ -537,6 +540,10 @@ private void cacheBlockData(
         cache -> blockAccessList.ifPresent(t -> cache.put(block.getHash(), t)));
   }
 
+  private void cacheBlockHeader(final BlockHeader blockHeader) {
```

Comment:
for uniformity can you call this function at line 534, instead of repeating the same code there?